examples/devsite-demo2/script.js

```diff
@@ -0,0 +1,349 @@
+  if (m) {
+    if (LANGUAGE_NAME[m[1]]) {
+      language = m[1];
+    }
+  }
+  // Load Blockly's language strings.
+  document.write(
+      '<script src="./node_modules/blockly/msg/' + language + '.js"></script>\n'
+  );
+})();
+
+/**
+ * Initialize the page once everything is loaded.
+ */
+function init() {
+  // Sort languages alphabetically.
+  const languages = [];
+  for (const lang in LANGUAGE_NAME) {
+    languages.push([LANGUAGE_NAME[lang], lang]);
+  }
+  /**
+   *
+   * @param a
+   * @param b
+   */
+  function comp_(a, b) {
+    // Sort based on first argument ('English', 'Русский', '简体字', etc).
+    if (a[0] > b[0]) return 1;
+    if (a[0] < b[0]) return -1;
+    return 0;
+  }
+  languages.sort(comp_);
+  // Populate the language selection dropdown.
+  const languageMenu = document.getElementById('languageDropdown');
+  for (let i = 0; i < languages.length; i++) {
+    const tuple = languages[i];
+    const lang = tuple[tuple.length - 1];
+    const option = new Option(tuple[0], lang);
+    if (lang === language) {
+      option.selected = true;
+    }
+    languageMenu.options.add(option);
+  }
+
+  // Changing languages involves reloading the page.  To not lose the blocks,
+  // they were stored in sessionStorage.  Here we retrieve that data.
+  let loadOnce = null;
+  try {
+    loadOnce = window.sessionStorage.getItem('loadOnceBlocks');
+    window.sessionStorage.removeItem('loadOnceBlocks');
+    loadOnce = JSON.parse(loadOnce);
+  } catch (e) {
+    // Storage can be flakey.
+    console.log(e);
+  }
+
+  // Inject localized category names.
+  toolboxJson['contents'][0].name = getMsg('Logic');
+  toolboxJson['contents'][1].name = getMsg('Loops');
+  toolboxJson['contents'][2].name = getMsg('Math');
+  toolboxJson['contents'][3].name = getMsg('Text');
+  toolboxJson['contents'][4].name = getMsg('Lists');
+  toolboxJson['contents'][5].name = getMsg('Colour');
+  // Separator.
+  toolboxJson['contents'][7].name = getMsg('Variables');
+  toolboxJson['contents'][8].name = getMsg('Procedures');
+
+  // Inject default variable name.
+  // https://github.com/google/blockly/issues/5238
+  let toolboxString = JSON.stringify(toolboxJson);
+  toolboxString = toolboxString.replace(
+      /%\{BKY_VARIABLES_DEFAULT_NAME\}/g,
+      Blockly.Msg.VARIABLES_DEFAULT_NAME
+  );
+  toolboxJson = JSON.parse(toolboxString);
+
+  // Inject Blockly.
+  const workspace = Blockly.inject('blocklyDiv', {
+    media: './node_modules/blockly/media/',
+    toolbox: toolboxJson,
+    rtl: LANGUAGE_RTL.includes(language),
+    zoom: {
+      controls: true,
+      maxScale: 1.8,
+      minScale: 0.6,
+      scaleSpeed: 1.2,
+      pinch: true,
+    },
+    trashcan: true,
+    theme: Blockly.Theme.defineTheme('modest', {
+      blockStyles: {
+        logic_blocks: {
+          colourPrimary: '#AECBFA',
+        },
+        loop_blocks: {
+          colourPrimary: '#A8DAB5',
+        },
+        math_blocks: {
+          colourPrimary: '#4285f4',
```

Comment:
Can we avoid repeating colors where one is just a lighter version of another? Blockly already uses light vs dark shades to have meaning, so it's confusing. It also implies to me some kind of relationship between the math and logic blocks that doesn't make sense. There are more options in the [extended Material color palette](https://m2.material.io/design/color/the-color-system.html#tools-for-picking-colors). Or you could also look at the [zelos theme](https://blockly-demo.appspot.com/static/tests/playgrounds/advanced_playground.html#renderer=thrasos&debugEnabled=false&theme=zelos) for a brighter set of default colors, though they aren't Google-themed.